plugin/pkg/scheduler/generic_scheduler.go

```diff
@@ -153,7 +154,9 @@ func findNodesThatFit(
 	nodeNameToInfo map[string]*schedulercache.NodeInfo,
 	nodes []*api.Node,
 	predicateFuncs map[string]algorithm.FitPredicate,
-	extenders []algorithm.SchedulerExtender) ([]*api.Node, FailedPredicateMap, error) {
+	extenders []algorithm.SchedulerExtender,
+	pFactory algorithm.MetadataProducer,
```

Comment:
nit: Can you call it metadataProducer instead pFactory (it would be more readable I think).